src/errors/runtime/index.js

```diff
@@ -1,34 +1,55 @@
 import { renderers } from 'callsite-record';
-import MESSAGE from './message';
+import TEMPLATES from './templates';
 import createStackFilter from '../create-stack-filter';
 import { getCallsiteForMethod } from '../get-callsite';
 import renderTemplate from '../../utils/render-template';
-
+import { RUNTIME_ERRORS } from '../types';
 
 const ERROR_SEPARATOR = '\n\n';
 
+class ProcessTemplateInstruction {
+    constructor (processFn) {
+        this.processFn = processFn;
+    }
+}
+
 // Errors
 export class GeneralError extends Error {
     constructor (...args) {
-        super(renderTemplate(...args));
+        const code     = args.shift();
+        const template = TEMPLATES[code];
+
+        super(renderTemplate(template, ...args));
+
+        Object.assign(this, { code, data: args });
         Error.captureStackTrace(this, GeneralError);
     }
 }
```

Comment:
I see a lot of calls to `renderTemplate` and `assign` methods. What if we create some `BaseError: Error` class with  `renderTemplate` and `assign` calls and couple of virtual method?

Reply:
I've tried.  Unfortunately, it does not simplified the code enough. So, I will leave it as is.